### ISSUE TITLE
VIDEORESIZE only issued once, further resizes are ignored

### ISSUE THREAD

**Issue by illume** [pygame/pygame#201]
Originally reported by: **Touch Tone (Bitbucket: [TouchTone](https://bitbucket.org/TouchTone), GitHub: [TouchTone](https://github.com/TouchTone))**

----------------------------------------

When using a RESIZABLE mode, the VIDEORESIZE event is only sent once, any further window resizes are ignored.

To reproduce: just run the VIDEORESIZE example (http://www.pygame.org/wiki/WindowResizing?parent=CookBook) and resize the window twice.

----------------------------------------
- Bitbucket: https://bitbucket.org/pygame/pygame/issue/201

**Comment by donkirkby** [comment]
I found the same problem on Ubuntu 16.04. Maximizing and edges seem to work fine, but the corner gets disconnected from the edges or something.

Here's a sample program:

    from logging import basicConfig, DEBUG, getLogger
    
    import pygame
    
    
    def draw(surface):
        x = surface.get_width()
        y = surface.get_height()
        surface.fill((50, 50, 50))
        surface.fill((200, 200, 200), (x/4, y/4, x/2, y/2))
    
    
    def resize(size):
        return pygame.display.set_mode(
            size,
            pygame.HWSURFACE | pygame.DOUBLEBUF | pygame.RESIZABLE)
    
    def main():
        basicConfig(level=DEBUG, format='%(asctime)s %(message)s')
        logger = getLogger()
        logger.info('Starting.')
        pygame.init()
        surface = resize((250, 150))
        draw(surface)
        pygame.display.flip()
        while True:
            event = pygame.event.wait()
            logger.debug('event %r', event)
            if event.type == pygame.QUIT:
                return
            elif event.type == pygame.VIDEORESIZE:
                surface = resize(event.size)
                draw(surface)
                pygame.display.flip()
    
    
    main()

If I run the program, then drag the corner, I see the following. Note that the last message happens when I briefly "reconnect" the edge by dragging it back past the previous boundary. It's only a brief connection, though. I can't get resizing to work properly again after it disconnects.

```
2017-11-17 22:37:30,202 Starting.
2017-11-17 22:37:30,365 event <Event(1-ActiveEvent {'gain': 0, 'state': 1})>
2017-11-17 22:37:37,713 event <Event(16-VideoResize {'size': (250, 151), 'w': 250, 'h': 151})>
2017-11-17 22:37:37,815 event <Event(16-VideoResize {'size': (250, 152), 'w': 250, 'h': 152})>
2017-11-17 22:37:37,937 event <Event(16-VideoResize {'size': (250, 153), 'w': 250, 'h': 153})>
2017-11-17 22:37:38,040 event <Event(16-VideoResize {'size': (250, 154), 'w': 250, 'h': 154})>
2017-11-17 22:37:38,151 event <Event(16-VideoResize {'size': (250, 157), 'w': 250, 'h': 157})>
2017-11-17 22:37:38,163 event <Event(16-VideoResize {'size': (251, 157), 'w': 251, 'h': 157})>
2017-11-17 22:37:38,173 event <Event(16-VideoResize {'size': (251, 158), 'w': 251, 'h': 158})>
2017-11-17 22:37:38,245 event <Event(16-VideoResize {'size': (252, 158), 'w': 252, 'h': 158})>
2017-11-17 22:37:38,297 event <Event(16-VideoResize {'size': (252, 159), 'w': 252, 'h': 159})>
2017-11-17 22:37:59,041 event <Event(16-VideoResize {'size': (252, 993), 'w': 252, 'h': 993})>

```

**Comment by LeonLenclos** [comment]
Same issue with Gnome on Debian 9.
Maximizing work but edges and corner resizing does not.

**Comment by Sigma-One** [comment]
Same issue on Fedora 27 with both i3wm and Gnome.

**Comment by PyRectangle** [comment]
I had the same issue on ubuntu 18.04. I fixed it through installing pygame 1.9.4.dev0.
```pip install pygame==1.9.4.dev0```

**Comment by kargaroc** [comment]
I can confirm that this happens on Arch Linux

**Comment by poikilos** [comment]
happens on recently updated Fedora 28 KDE with python3-pygame 1.9.3

**Comment by rubenvereecken** [comment]
I have the same problem on Linux Mint, even with `pygame==1.9.4.dev0`. Both after maximizing and regular resizing. The event only stops activating when I actually `pygame.display.set_mode` with the new display size.

**Comment by PyRectangle** [comment]
@rubenvereecken I could reproduce your issue on Linux Mint 19.
I could fix it through building the [1.9.4.3 branch](https://github.com/pygame/pygame/tree/1.9.4.3).

**Comment by illume** [comment]
It's possible Arch Linux (and other distros) are patching SDL with a fix for this.
https://git.archlinux.org/svntogit/packages.git/tree/trunk?h=packages/sdl

I'll make an issue to include all their patches (and the custom ones Debian and Ubuntu do). Additionally, I've asked about SDL1.2 having a bug fix release, and there is some interest (and more importantly no opposition). So I will work on gathering all these bugfix patches and try to get them included in this bugfix release.

Hopefully once we include all the various SDL related patches in the 'manylinux' pygame builds this will be sorted out. I guess it could also be an issue with the manylinux builds.

If anyone wants to try to help and fix this, the readme file in `buildconfig/manylinux` describes how we make the linux binary wheels in great detail. It's automated. So the task would be to downloading and applying the different patches and then testing the wheels.

**Comment by nthykier** [comment]
I tried testing this bug with the code listed in https://github.com/pygame/pygame/issues/201#issuecomment-345423040 and:

- pygame 1.9.6 (Debian's version / SDL 1.2): **cannot** reproduce the bug
- pygame 2 (master branch / SDL 2): Program crashes with `pygame.error: Window surface is invalid, please call SDL_GetWindowSurface() to get a new surface`


Full log of the crash:

```
pygame 2.0.0.dev7 (SDL 2.0.10, python 3.7.7)
Hello from the pygame community. https://www.pygame.org/contribute.html
2020-03-22 20:38:34,607 Starting.
2020-03-22 20:38:35,147 event <Event(4352-Unknown {})>
2020-03-22 20:38:35,147 event <Event(4352-Unknown {})>
2020-03-22 20:38:35,148 event <Event(4352-Unknown {})>
2020-03-22 20:38:35,148 event <Event(32770-VideoExpose {})>
2020-03-22 20:38:35,148 event <Event(32768-ActiveEvent {'gain': 1, 'state': 1, 'window': None})>
2020-03-22 20:38:36,175 event <Event(768-KeyDown {'unicode': '', 'key': 1073742050, 'mod': 4352, 'scancode': 226, 'window': None})>
2020-03-22 20:38:36,257 event <Event(32770-VideoExpose {})>
2020-03-22 20:38:36,278 event <Event(769-KeyUp {'key': 1073742050, 'mod': 4096, 'scancode': 226, 'window': None})>
2020-03-22 20:38:36,278 event <Event(32768-ActiveEvent {'gain': 0, 'state': 1, 'window': None})>
2020-03-22 20:40:49,670 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:49,670 event <Event(32768-ActiveEvent {'gain': 1, 'state': 1, 'window': None})>
2020-03-22 20:40:49,670 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,377 event <Event(32769-VideoResize {'size': (251, 150), 'w': 251, 'h': 150})>
2020-03-22 20:40:52,379 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,379 event <Event(32769-VideoResize {'size': (252, 150), 'w': 252, 'h': 150})>
2020-03-22 20:40:52,381 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,381 event <Event(32769-VideoResize {'size': (253, 150), 'w': 253, 'h': 150})>
2020-03-22 20:40:52,388 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,388 event <Event(32769-VideoResize {'size': (254, 150), 'w': 254, 'h': 150})>
2020-03-22 20:40:52,391 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,391 event <Event(32769-VideoResize {'size': (256, 151), 'w': 256, 'h': 151})>
2020-03-22 20:40:52,393 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,428 event <Event(32769-VideoResize {'size': (260, 152), 'w': 260, 'h': 152})>
2020-03-22 20:40:52,433 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,448 event <Event(32769-VideoResize {'size': (263, 153), 'w': 263, 'h': 153})>
2020-03-22 20:40:52,453 event <Event(32770-VideoExpose {})>
2020-03-22 20:40:52,477 event <Event(32769-VideoResize {'size': (264, 154), 'w': 264, 'h': 154})>
Traceback (most recent call last):
  File "dist-packages/foo.py", line 37, in <module>
    main()
  File "dist-packages/foo.py", line 34, in main
    pygame.display.flip()
pygame.error: Window surface is invalid, please call SDL_GetWindowSurface() to get a new surface
```

**Comment by nthykier** [comment]
On the topic of distribution specific patches to SDL 1.2, then Debian carries a patch with the interesting name `fix_window_resizing.diff` (https://salsa.debian.org/sdl-team/libsdl1.2/-/blob/master/debian/patches/fix_window_resizing.diff)

I strongly suspect the issue with the resizing is a sdl 1.2 bug rather than a pygame bug.  That said, pygame 2/SDL2 emits an interesting result that might be worth a different bug.

**Comment by MyreMylar** [comment]
This should now be fixed everywhere using 2.0.0.dev10 and the new way of handling resizing windows - as visible on the wiki or right here:

    import pygame
    from pygame.locals import *
    pygame.init()
    screen = pygame.display.set_mode((500, 500), RESIZABLE)
    pic = pygame.image.load("example.png")  # You need an example picture in the same folder as this file!
    running = True
    while running:
        pygame.event.pump()
        event = pygame.event.wait()
        if event.type == QUIT:
            running = False
        elif event.type == VIDEORESIZE:
            screen.blit(pygame.transform.scale(pic, event.dict['size']), (0, 0))
            pygame.display.update()
        elif event.type == VIDEOEXPOSE:  # handles window minimising/maximising
            screen.fill((0, 0, 0))
            screen.blit(pygame.transform.scale(pic, screen.get_size()), (0, 0))
            pygame.display.update() 

I've just tested it on a raspberry Pi with Debian/Raspberry Pi OS (who knew they renamed that from Raspbian?) and it works well now. Events continue to get generated.

I'll close this for now, as I think the new method shouldn't require any distro hacks but feel free to re-open if you find a specific distro where the code above and a pygame later than 2.0.0.dev10 doesn't work.

**Comment by donkirkby** [comment]
Thanks for the fix, this now works for me with pygame 2.0.0.dev10 and Python 3.8.3 on Ubuntu 18.04.

When I tried rerunning my sample code from [2017, above], it needed a slight tweak to get working. In [pygame 2], you don't need to resize the surface after receiving the resize event.

    from logging import basicConfig, DEBUG, getLogger
    
    import pygame
    
    
    def draw(surface):
        x = surface.get_width()
        y = surface.get_height()
        surface.fill((50, 50, 50))
        surface.fill((200, 200, 200), (x/4, y/4, x/2, y/2))
    
    
    def resize(size):
        return pygame.display.set_mode(
            size,
            pygame.HWSURFACE | pygame.DOUBLEBUF | pygame.RESIZABLE)
    
    
    def main():
        basicConfig(level=DEBUG, format='%(asctime)s %(message)s')
        logger = getLogger()
        logger.info('Starting.')
        pygame.init()
        surface = resize((250, 150))
        draw(surface)
        pygame.display.flip()
        while True:
            event = pygame.event.wait()
            logger.debug('event %r', event)
            if event.type == pygame.QUIT:
                return
            elif event.type == pygame.VIDEORESIZE:
                draw(surface)
                pygame.display.flip()
    
    
    main()

Fun fact: you can shrink the width to zero and get an invisible window! I guess you could check the new size on the event and force a resize if it gets too big or too small.

[2017, above]: https://github.com/pygame/pygame/issues/201#issuecomment-345423040
[pygame 2]: http://www.pygame.org/wiki/WindowResizing
